cmd/ci-operator-checkconfig/main.go

```diff
@@ -27,28 +30,57 @@ func (o *options) parse() error {
 	var registryDir string
 	flag.StringVar(&o.configDir, "config-dir", "", "The directory containing configuration files.")
 	flag.StringVar(&registryDir, "registry", "", "Path to the step registry directory")
+	flag.IntVar(&o.maxConcurrency, "concurrency", 0, "Maximum number of concurrent in-flight goroutines.")
 	flag.Parse()
 	if o.configDir == "" {
 		return errors.New("The --config-dir flag is required but was not provided")
 	}
+	if o.maxConcurrency == 0 {
+		o.maxConcurrency = runtime.GOMAXPROCS(0)
+	}
 	if err := o.loadResolver(registryDir); err != nil {
 		return fmt.Errorf("failed to load registry: %w", err)
 	}
 	return nil
 }
 
 func (o *options) validate() (ret []error) {
-	seen := tagSet{}
-	if err := config.OperateOnCIOperatorConfigDir(o.configDir, func(configuration *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
-		// basic validation of the configuration is implicit in the iteration
-		if err := o.validateConfiguration(seen, configuration, repoInfo); err != nil {
-			ret = append(ret, fmt.Errorf("error validating configuration %s: %w", repoInfo.Filename, err))
+	type workItem struct {
+		configuration *api.ReleaseBuildConfiguration
+		repoInfo      *config.Info
+	}
+	ch, errCh := make(chan workItem), make(chan error)
+	seen := make([]tagSet, o.maxConcurrency)
```

Comment:
IMO this would make more sense as a channel to funnel data back? Passing pointers around to concurrent programs for mutation is a good way to get races :)